src/windows/PushwooshPluginProxy.js

```diff
@@ -29,7 +34,8 @@ module.exports = {
     registerDevice: function (success, fail) {
         if (!this.service) {
             // postpone
-            setTimeout(function () { this.registerDevice(success, fail) }, 1000);
+            setTimeout(function () { this.plugins.pushNotification.registerDevice(success, fail); }, 5000);

```

Comment:
I think it's still a little unclear to use `this` inside of `setTimeout()`, since it's not obvious what `this` is in that case. It seems like most other places in this file are using `cordova.require()`, like so:

```js
setTimeout(function() {
    cordova.require('pushwoosh-cordova-plugin.PushNotification').registerDevice(success, fail);
}, 0);
```

Can we follow that pattern?

Reply:
That pattern won't work in this case because registerDevice is defined in the current proxy file, not in PushNotification.

Reply:
Files ticket MOBILE-850 to track this.

Reply:
Great, thanks @renebarranco!